[#10322] Rich text editor breaks when moved during reorder

jtankw3 opened the pull request:
Fixes #10322

This issue is talked about here: https://stackoverflow.com/questions/2535569/tinymce-editor-dislikes-being-moved-around
I believe the only solution is to remove the TinyMCE before moving the elements and adding it back after.

However, I do not think it is possible to have a listener for the editor is called during when the editor is moved. 
It is also difficult for `instructor-session-edit-page.component` to interact with the  editor directly.

Hence, I believe the best solution is to reload all the questions,  thus reloading the editor as well.

**Outline of Solution**
- Change from moving `questionForms` to reloading on reorder

ccyccyccy commented:
LGTM. I don't see a cleaner way.

xpdavid commented on src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts:
```diff
@@ -538,24 +538,21 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
 
   /**
-   * Moves question edit form from the original position to the new position.
+   * Reloads feedback questions.
```
I think we need to explain the workaround a bit as this violate our design in frontend where we want to minimize the unnecessary communication.

wkurniawan07 commented:
@jtankw3 good job for noticing this behaviour. However, there is a better solution:
- Add `ChangeDetectorRef` to the session edit page's component
- After re-ordering, the question at the new position needs to have `isCollapsed = true`. This will, on paper, destroy the edit components including the rich text editor.
- Trigger the `ChangeDetectorRef` manually (using `detectChanges`) so that the previously mentioned action will take place
- Change back the `isCollapsed = false`

jtankw3 commented:
> @jtankw3 good job for noticing this behaviour. However, there is a better solution:
> 
> * Add `ChangeDetectorRef` to the session edit page's component
> * After re-ordering, the question at the new position needs to have `isCollapsed = true`. This will, on paper, destroy the edit components including the rich text editor.
> * Trigger the `ChangeDetectorRef` manually (using `detectChanges`) so that the previously mentioned action will take place
> * Change back the `isCollapsed = false`

Thank you for the direction for solving this problem.
- I realized that if a question is moved down the form, all the question forms that are pushed up will also be affected by this problem. Hence I have implemented the above for all question models that are shifted up after any reorder.
- I also added a check to prevent originally collapsed questions from being expanded.

wkurniawan07 commented on src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts:
```diff
@@ -556,6 +557,15 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
+      .filter((model: QuestionEditFormModel) => !model.isCollapsed);
+    movedExpandedQuestions.forEach((model: QuestionEditFormModel) => model.isCollapsed = true);
+    this.changeDetectorRef.detectChanges();
+    movedExpandedQuestions.forEach((model: QuestionEditFormModel) => model.isCollapsed = false);
```
This will make all modals previously collapsed to be expanded. Only those which are previously expanded should be re-expanded.

jtankw3 replied:
@wkurniawan07  I have filtered to only those that are not collapsed so those that are originally collapsed will not be affected. (line 565)

wkurniawan07 replied:
Ah right, I missed that.
